std-bits/table/src/main/java/org/enso/table/data/column/operation/map/SpecializedIsInOp.java

```diff
@@ -0,0 +1,101 @@
+package org.enso.table.data.column.operation.map;
+
+import java.util.BitSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.function.Function;
+import org.enso.base.polyglot.Polyglot_Utils;
+import org.enso.table.data.column.storage.BoolStorage;
+import org.enso.table.data.column.storage.Storage;
+
+/**
+ * A specialized implementation for the IS_IN operation for builtin types, relying on hashing. Since
+ * for some columns we know what types of objects can be stored, we can filter out any objects that
+ * do not match that type and then rely on a consistent definition of hashcode for these builtin
+ * types (which is not available in general for custom objects).
+ */
+public class SpecializedIsInOp<T extends Storage> extends MapOperation<T> {
+  /**
+   * An optimized representation of the vector of values to match.
+   *
+   * <p>It indicates whether the vector contained a null value and contains a hashmap of the vector
+   * elements for faster contains checks.
+   */
+  public record CompactRepresentation(HashSet<Object> coercedValues, boolean hasNulls) {}
```

Comment:
Could we do a generic version? I don't think will make any real difference but wondered if nicer to have `HashSet<String>` etc for example.

Reply:
I tried. It was a bit more complex than we'd like, but maybe the changes could open venues for writing more type-aware operations in the future.

I first made the `SpecializedIsInOp` generic as asked ( 2430d469d859c07cb5b1ea28ed7e05f3ceb64c72 ), to do so I needed to introduce some type markers to make stuff compile.

Then in 5eb9c3f86d638c84936382701946354e183200dd I have removed the temporarily added `TypedStorage<T>` marker interface and _just_ made `Storage` know its type `Storage<T>`. This required some more work though.

Please tell me if you think it's worth keeping both of these changes or only one of them or none - I'll amend as necessary and then get the PR merged finally.